src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -1297,6 +1429,21 @@ otError RadioSpinel<InterfaceType, ProcessContextType>::Get(spinel_prop_key_t aK
     va_end(mPropertyArgs);
     mPropertyFormat = nullptr;
 
+#if OPENTHREAD_SPINEL_CONFIG_RCP_RESTORATION_ENABLE
+    if (mRcpFailed)
```

Comment:
Do we need to check `mRcpFailed` **before** the first `RequestV` ?

Reply:
It still works fine without checking before requesting, the first attempt will fail. Starting recovery earlier is good but also requires another block of code added to every function of get/set/insert/remove. Considering https://github.com/openthread/openthread/pull/5694#discussion_r517845367, is it better?

Reply:
After a second thought I think it's unnecessary. Because the only places could set `mRcpFailed` are in `Process` and get/set/insert/remove operations, `RecoverFromRcpFailure` is called in each of them, so failures will be immediately recovered.